tests/test_collaborativerecommender.py

```diff
@@ -51,22 +90,39 @@ def test_can_recommend_no_model(activate_error_responses):
 
 
 @responses.activate
-def test_recommendations(activate_responses):
+def test_empty_recommendations(activate_responses):
     # Tests that the empty recommender always recommends an empty list
-    # of addons.
+    # of addons if we have no addons
     r = CollaborativeRecommender()
+
+    assert not r.can_recommend({})
     recommendations = r.recommend({}, 1)
+    assert isinstance(recommendations, list)
+    assert len(recommendations) == 0
+
 
+@responses.activate
+def test_best_recommendation(activate_responses):
     # Make sure the structure of the recommendations is correct and that we
     # recommended the the right addon.
+    r = CollaborativeRecommender()
+
+    # An empty set of addons should always give an empty list of
```

Comment:
The "empty set" case is already covered by `test_empty_recommendations`: should we drop lines 110-114?

Reply:
You're right.  I'm dropping lines 110-114